libshaderc_util/CMakeLists.txt

```diff
@@ -34,8 +34,8 @@ endif(SHADERC_ENABLE_INSTALL)
 
 find_package(Threads)
 target_link_libraries(shaderc_util PRIVATE
-  glslang OSDependent OGLCompiler HLSL glslang SPIRV
-  SPIRV-Tools-opt ${CMAKE_THREAD_LIBS_INIT})
+  glslang OSDependent OGLCompiler glslang HLSL SPIRV
+  SPIRV-Tools-opt SPIRV-Tools ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
CMake already has a "PUBLIC" dependency from SPIRV-Tools-opt to SPIRV-Tools.  So it seems this is not necessary?

What error are you trying to fix here?

Reply:
Correct me if I'm wrong, but those dependencies only do anything if you actually build SPIRV-Tools-opt / SPIRV-Tools using your own cmake scripts, from third_party. But in practice, I am not building libshaderc against the third_party versions - I am building them against the system-wide SPIRV-Tools installation. (As the commit message points out)

The concrete issue being solved here is https://github.com/sjnewbury/gentoo-gpu/issues/22, which prevents `libshaderc` from working on Gentoo. The gentoo packaging policy forbids the use of vendored `third_party`-style libraries in favor of system-wide versions of those libraries wherever avoidable, and this is such a case.

I can either ship the work-around from this commit as part of the gentoo package, or I can try submitting it upstream - which is what I tried here.